java-spiffe-helper/Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM eclipse-temurin:17-jdk AS builder
```

Comment:
I suggest moving this Dockerfile for `java-spiffe-helper` to the project's root. This simplifies build context management and path referencing, making the build process more straightforward.

Reply:
Suggestions for Dockerfile:

- Cache Gradle Dependencies: Separate Gradle config files from the rest of the code to improve caching, and run two separate gradle commands (dependencies and build).
- Non-Root User: Consider creating a non-root user in the runner stage.
- Explicit File Copying: Instead of copying the entire project context (COPY . /builder), explicitly copy only what's necessary.
- Versioning Strategy: Rethink including `JAVA_SPIFFE_HELPER_VERSION` in the Dockerfile. Tagging the Docker image with the project version might be more efficient.

Reply:
Explicit File Copying: I don't see any benefit here, except build time probably. 1. The context file won't end up in the final image. 2. Every change to the dependencies of `java-spiffe-helper` would require an adjustment of the `Dockerfile` as well.
Versioning Strategy: Yeah, my issue here was that the name of the built `Jar` is dependent on the version. Solved this now by setting some special version during `docker build`. Easier to find and copy then.